src/string.c

```diff
@@ -174,14 +174,25 @@ char *stripslashes(char *str)
 * printf("_%s_\n", s);
 * \endcode
 */
-void ltrim(char *str)
+char * cgi_ltrim(char *str)
 {
-	char *s = str;
+    char *s = str;
 
-	while (isspace(*s))
-		s++;
+    /* nothing to do if str is NULL or zero-length */
+	if (str && *str)
+	{
+		/* find first non-space character */
+		while (isspace(*s))
+			s++;
+
+		/* if any space characters were found, remove them by copying from
+		 * beyond them to the beginning of the source string
+		 */
+		if (s > str)
+			while ((*str++ = *s++));
+	}
 
-	while ((*str++ = *s++));
+	return str;
```

Comment:
This will return a wrong pointer. Because you copy the bytes to the front in the loop above, you really want to return the original pointer, but you modify it in the loop! So the returned pointer does not point to the beginning of the string but somewhere in between, which is probably not what you want.